### PR TITLE
*: change package name from `plan` to `planner`

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

This PR makes the following two package changes:
1. change the "plan" package to "planner"
2. move all the source files inside the "planner" package to the "planner/core" package

The propose of this PR is to:
1. modularize the "planner" package
2. prevent the chaos to implement the cascades planner.

### What is changed and how it works?

almost all the code changes are made by the shell scripts

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @eurekaka @XuHuaiyu PTAL

**Comment by jackysp** [comment]
Please update the PR title.

File: executor/builder.go [hunk 1]
```diff
@@ -1738,27 +1738,27 @@ func (b *executorBuilder) buildIndexLookUpReader(v *plan.PhysicalIndexLookUpRead
 // dataReaderBuilder build an executor.
 // The executor can be used to read data in the ranges which are constructed by datums.
 // Differences from executorBuilder:
-// 1. dataReaderBuilder calculate data range from argument, rather than plan.
+// 1. dataReaderBuilder calculate data range from argument, rather than core.
```
Comment (eurekaka) [inline executor/builder.go]: false positive

File: executor/builder.go [hunk 2]
```diff
@@ -1821,7 +1821,7 @@ func (builder *dataReaderBuilder) buildIndexLookUpReaderForIndexJoin(ctx context
 	return e, errors.Trace(err)
 }
 
-// buildKvRangesForIndexJoin builds kv ranges for index join when the inner plan is index scan plan.
+// buildKvRangesForIndexJoin builds kv ranges for index join when the inner plan is index scan core.
```
Comment (eurekaka) [inline executor/builder.go]: ditto

File: executor/union_scan.go [hunk 3]
```diff
@@ -103,7 +103,7 @@ type UnionScanExec struct {
 	conditions []expression.Expression
 	columns    []*model.ColumnInfo
 
-	// belowHandleIndex is the handle's position of the below scan plan.
+	// belowHandleIndex is the handle's position of the below scan planner.
```
Comment (eurekaka) [inline executor/union_scan.go]: ditto

File: expression/expr_to_pb.go [hunk 4]
```diff
@@ -66,7 +66,7 @@ func ExpressionsToPB(sc *stmtctx.StatementContext, exprs []Expression, client kv
 	return
 }
 
-// ExpressionsToPBList converts expressions to tipb.Expr list for new plan.
+// ExpressionsToPBList converts expressions to tipb.Expr list for new planner.
```
Comment (eurekaka) [inline expression/expr_to_pb.go]: ditto

File: planner/core/common_plans.go [hunk 5]
```diff
@@ -102,14 +102,14 @@ type ChecksumTable struct {
 	Tables []*ast.TableName
 }
 
-// CancelDDLJobs represents a cancel DDL jobs plan.
+// CancelDDLJobs represents a cancel DDL jobs planner.
```
Comment (eurekaka) [inline planner/core/common_plans.go]: ditto

File: planner/core/common_plans.go [hunk 6]
```diff
@@ -102,14 +102,14 @@ type ChecksumTable struct {
...
 type CancelDDLJobs struct {
 	baseSchemaProducer
 
 	JobIDs []int64
 }
 
-// Prepare represents prepare plan.
+// Prepare represents prepare planner.
 type Prepare struct {
```
Comment (eurekaka) [inline planner/core/common_plans.go]: ditto

File: planner/core/common_plans.go [hunk 7]
```diff
@@ -393,21 +393,21 @@ type LoadData struct {
...
+// DDL represents a DDL statement planner.
 type DDL struct {
 	baseSchemaProducer
 
 	Statement ast.DDLNode
 }
 
-// Explain represents a explain plan.
+// Explain represents a explain planner.
 type Explain struct {
```
Comment (eurekaka) [inline planner/core/common_plans.go]: ditto

File: planner/core/exhaust_physical_plans.go [hunk 8]
```diff
@@ -641,7 +641,7 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *property.PhysicalProperty) ([]Phys
 }
 
 // LogicalJoin can generates hash join, index join and sort merge join.
-// Firstly we check the hint, if hint is figured by user, we force to choose the corresponding physical plan.
+// Firstly we check the hint, if hint is figured by user, we force to choose the corresponding physical planner.
```
Comment (eurekaka) [inline planner/core/exhaust_physical_plans.go]: ditto

File: planner/core/find_best_task.go [hunk 9]
```diff
@@ -129,7 +129,7 @@ func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTas
 			continue
 		}
 
-		// combine best child tasks with parent physical plan.
+		// combine best child tasks with parent physical planner.
```
Comment (eurekaka) [inline planner/core/find_best_task.go]: ditto

File: planner/core/find_best_task.go [hunk 10]
```diff
@@ -201,7 +201,7 @@ func (ds *DataSource) tryToGetDualTask() (task, error) {
 func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err error) {
 	// If ds is an inner plan in an IndexJoin, the IndexJoin will generate an inner plan by itself.
 	// So here we do nothing.
-	// TODO: Add a special prop to handle IndexJoin's inner plan.
+	// TODO: Add a special prop to handle IndexJoin's inner planner.
 	// Then we can remove forceToTableScan and forceToIndexScan.
```
Comment (eurekaka) [inline planner/core/find_best_task.go]: ditto

File: planner/core/logical_plan_builder.go [hunk 11]
```diff
@@ -303,7 +303,7 @@ func (b *planBuilder) buildJoin(joinNode *ast.Join) (LogicalPlan, error) {
 		joinPlan.JoinType = InnerJoin
 	}
 
-	// Merge sub join's redundantSchema into this join plan. When handle query like
+	// Merge sub join's redundantSchema into this join planner. When handle query like
```
Comment (eurekaka) [inline planner/core/logical_plan_builder.go]: ditto

File: planner/core/logical_plans.go [hunk 12]
```diff
@@ -91,7 +91,7 @@ const (
 	preferMergeJoin
 )
 
-// LogicalJoin is the logical join plan.
+// LogicalJoin is the logical join planner.
```
Comment (eurekaka) [inline planner/core/logical_plans.go]: ditto

File: planner/core/logical_plans.go [hunk 13]
```diff
@@ -587,20 +587,20 @@ type LogicalTopN struct {
...
+// LogicalLimit represents offset and limit planner.
 type LogicalLimit struct {
 	baseLogicalPlan
 
 	Offset uint64
 	Count  uint64
 }
 
-// LogicalLock represents a select lock plan.
+// LogicalLock represents a select lock planner.
```
Comment (eurekaka) [inline planner/core/logical_plans.go]: ditto

File: planner/core/physical_plans.go [hunk 14]
```diff
@@ -82,7 +82,7 @@ type PhysicalIndexLookUpReader struct {
 	tablePlan  PhysicalPlan
 }
 
-// PhysicalIndexScan represents an index scan plan.
+// PhysicalIndexScan represents an index scan planner.
```
Comment (eurekaka) [inline planner/core/physical_plans.go]: ditto

File: planner/core/physical_plans.go [hunk 15]
```diff
@@ -270,7 +270,7 @@ type PhysicalUnionAll struct {
 	physicalSchemaProducer
 }
 
-// AggregationType stands for the mode of aggregation plan.
+// AggregationType stands for the mode of aggregation planner.
```
Comment (eurekaka) [inline planner/core/physical_plans.go]: ditto

File: planner/core/plan.go [hunk 16]
```diff
@@ -40,7 +40,7 @@ type Plan interface {
 
 	context() sessionctx.Context
 
-	// property.StatsInfo will return the property.StatsInfo for this plan.
+	// property.StatsInfo will return the property.StatsInfo for this planner.
```
Comment (eurekaka) [inline planner/core/plan.go]: ditto

File: planner/core/plan.go [hunk 17]
```diff
@@ -127,13 +127,13 @@ type PhysicalPlan interface {
...
 
-	// StatsCount returns the count of property.StatsInfo for this plan.
+	// StatsCount returns the count of property.StatsInfo for this planner.
 	StatsCount() float64
 
 	// Get all the children.
 	Children() []PhysicalPlan
 
-	// SetChildren sets the children for the plan.
+	// SetChildren sets the children for the planner.
```
Comment (eurekaka) [inline planner/core/plan.go]: ditto

File: planner/core/plan_to_pb.go [hunk 18]
```diff
@@ -163,7 +163,7 @@ func SetPBColumnsDefaultValue(ctx sessionctx.Context, pbColumns []*tipb.ColumnIn
 // SupportStreaming returns true if a pushed down operation supports using coprocessor streaming API.
 // Note that this function handle pushed down physical plan only! It's called in constructDAGReq.
 // Some plans are difficult (if possible) to implement streaming, and some are pointless to do so.
-// TODO: Support more kinds of physical plan.
+// TODO: Support more kinds of physical planner.
```
Comment (eurekaka) [inline planner/core/plan_to_pb.go]: ditto

File: planner/core/planbuilder.go [hunk 19]
```diff
@@ -1374,7 +1374,7 @@ func (b *planBuilder) buildDDL(node ast.DDLNode) Plan {
 	return p
 }
 
-// buildTrace builds a trace plan. Inside this method, it first optimize the
+// buildTrace builds a trace planner. Inside this method, it first optimize the
```
Comment (eurekaka) [inline planner/core/planbuilder.go]: ditto

File: planner/core/point_get_plan.go [hunk 20]
```diff
@@ -108,7 +108,7 @@ func (p *PointGetPlan) Children() []PhysicalPlan {
 	return nil
 }
 
-// SetChildren sets the children for the plan.
+// SetChildren sets the children for the planner.
```
Comment (eurekaka) [inline planner/core/point_get_plan.go]: ditto

File: planner/core/rule_decorrelate.go [hunk 21]
```diff
@@ -111,7 +111,7 @@ func (la *LogicalApply) deCorColFromEqExpr(expr expression.Expression) expressio
 	return nil
 }
 
-// decorrelateSolver tries to convert apply plan to join plan.
+// decorrelateSolver tries to convert apply plan to join planner.
```
Comment (eurekaka) [inline planner/core/rule_decorrelate.go]: ditto

File: planner/core/rule_eliminate_projection.go [hunk 22]
```diff
@@ -105,7 +105,7 @@ func (pe *projectionEliminater) optimize(lp LogicalPlan) (LogicalPlan, error) {
 	return root, nil
 }
 
-// eliminate eliminates the redundant projection in a logical plan.
+// eliminate eliminates the redundant projection in a logical planner.
```
Comment (eurekaka) [inline planner/core/rule_eliminate_projection.go]: ditto

File: planner/core/task.go [hunk 23]
```diff
@@ -43,7 +43,7 @@ type copTask struct {
 	indexPlan PhysicalPlan
 	tablePlan PhysicalPlan
 	cst       float64
-	// indexPlanFinished means we have finished index plan.
+	// indexPlanFinished means we have finished index planner.
```
Comment (eurekaka) [inline planner/core/task.go]: ditto

File: planner/core/task.go [hunk 24]
```diff
@@ -321,11 +321,11 @@ func (p *PhysicalTopN) getPushedDownTopN() *PhysicalTopN {
...
 func (p *PhysicalTopN) attach2Task(tasks ...task) task {
 	t := tasks[0].copy()
-	// This is a topN plan.
+	// This is a topN planner.
 	if copTask, ok := t.(*copTask); ok && p.canPushDown() {
 		pushedDownTopN := p.getPushedDownTopN()
-		// If all columns in topN are from index plan, we can push it to index plan. Or we finish the index plan and
-		// push it to table plan.
+		// If all columns in topN are from index plan, we can push it to index planner. Or we finish the index plan and
+		// push it to table planner.
```
Comment (eurekaka) [inline planner/core/task.go]: ditto

File: planner/core/trace.go [hunk 25]
```diff
@@ -1,10 +1,10 @@
-package plan
+package core
 
 import (
 	"github.com/pingcap/tidb/ast"
 )
 
-// Trace represents a trace plan.
+// Trace represents a trace planner.
```
Comment (eurekaka) [inline planner/core/trace.go]: ditto

File: sessionctx/variable/session.go [hunk 26]
```diff
@@ -206,10 +206,10 @@ type SessionVars struct {
 	// ConnectionID is the connection id of the current session.
 	ConnectionID uint64
 
-	// PlanID is the unique id of logical and physical plan.
+	// PlanID is the unique id of logical and physical planner.
 	PlanID int
 
-	// PlanColumnID is the unique id for column when building plan.
+	// PlanColumnID is the unique id for column when building planner.
```
Comment (eurekaka) [inline sessionctx/variable/session.go]: ditto

File: util/ranger/checker.go [hunk 27]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/pingcap/tidb/types"
 )
 
-// conditionChecker checks if this condition can be pushed to index plan.
+// conditionChecker checks if this condition can be pushed to index planner.
 type conditionChecker struct {
```
Comment (eurekaka) [inline util/ranger/checker.go]: ditto

**Comment by zz-jason** [comment]
@winoros @eurekaka comments addressed and all the tests are passed, PTAL

**Comment by shenli** [comment]
@zz-jason Please resolve the conflicts.

**Comment by eurekaka** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu @shenli PTAL

**Pull request merged**